src/Drupal/DKANExtension/Context/WorkflowContext.php

```diff
@@ -30,23 +35,27 @@ public function transitionModerationState($named_entity, $state, $date = null) {
       throw new \Exception("State '$state' is not available. All possible states are [$possible_states].");
     }
 
-    $current_user = $this->getCurrentUser();
+    $current_user = $user = ($workflow_user) ? user_load_by_name($workflow_user) : $this->getCurrentUser();
```

Comment:
You have the possibility that you throw an exception, and never get to reset the user back. You should move this code to right before you need to call  workbench_moderation_moderate(); and then fix the user immediately after. Maybe even do that in a try catch finally block.